cmd/sops/main.go

```diff
@@ -443,6 +444,55 @@ func main() {
 				return nil
 			},
 		},
+		{
+			Name:      "filestatus",
+			Usage:     "check the status of the file, returning encryption status",
+			ArgsUsage: `file`,
+			Flags:     []cli.Flag{},
+			Action: func(c *cli.Context) error {
+				// _, err := config.FindConfigFile(".")
+				// if err != nil {
+				// 	return common.NewExitError(err, codes.ErrorGeneric)
+				// }
+				if c.NArg() < 1 {
+					return common.NewExitError("Error: no file specified", codes.NoFileSpecified)
+				}
+
+				fileName := c.Args()[0]
+
+				inputStore := inputStore(c, fileName)
+				outputStore := outputStore(c, fileName)
+
+				svcs := keyservices(c)
+				opts := encryptOpts{
+					Cipher:            aes.NewCipher(),
+					InputStore:        inputStore,
+					OutputStore:       outputStore,
+					InputPath:         fileName,
+					UnencryptedSuffix: "",
+					EncryptedSuffix:   "",
+					EncryptedRegex:    "",
+					KeyGroups:         []sops.KeyGroup{},
+					GroupThreshold:    0,
+					KeyServices:       svcs,
+				}
+
+				fileBytes, err := ioutil.ReadFile(opts.InputPath)
+				if err != nil {
+					return common.NewExitError(fmt.Sprintf("Error reading file: %s", err), codes.CouldNotReadInputFile)
+				}
+				branches, err := opts.InputStore.LoadPlainFile(fileBytes)
+				if err != nil {
+					return common.NewExitError(fmt.Sprintf("Error unmarshalling file: %s", err), codes.CouldNotReadInputFile)
+				}
+				if err := ensureNoMetadata(opts, branches[0]); err != nil {
+					return common.NewExitError("File is encrypted", codes.FileAlreadyEncrypted)
+				}
```

Comment:
This code is a duplicate of https://github.com/mozilla/sops/blob/4b5b7ae308839fb4bbd6cc35851f93c01322643b/cmd/sops/encrypt.go#L58-L69

Should this duplication be avoided? If so, what would be the best way?

Reply:
Yeah, I think we could extract that function to the `sops/cmd/common` package so that we can call it from both places.

Reply:
Never mind, I misread -- no need to deduplicate this.